sdk/formrecognizer/ai-form-recognizer/review/ai-form-recognizer.api.md

```diff
@@ -0,0 +1,741 @@
+}
+
+// @public (undocumented)
+export interface RawReceiptResult {
+    docType: "prebuilt:receipt";
+    fields: RawReceipt;
+    pageRange: number[];
+}
+
+// @public
+export interface ReadResult {
+    angle: number;
+    height: number;
+    language?: Language;
+    lines?: TextLine[];
+    page: number;
+    unit: LengthUnit;
+    width: number;
+}
+
+// @public (undocumented)
+export interface Receipt {
+    // (undocumented)
+    items: ReceiptItem[];
+    // (undocumented)
+    merchantAddress: string;
+    // (undocumented)
+    merchantName: string;
+    // (undocumented)
+    merchantPhoneNumber: string;
+    // (undocumented)
+    receiptType: string;
+    // (undocumented)
+    subtotal: number;
+    // (undocumented)
+    tax: number;
+    // (undocumented)
+    tip: number;
+    // (undocumented)
+    total: number;
+    // (undocumented)
+    transactionDate: string;
+    // (undocumented)
+    transactionTime: string;
+}
+
+// @public (undocumented)
+export interface ReceiptItem {
+    // (undocumented)
+    name?: string;
+    // (undocumented)
+    quantity?: number;
+    // (undocumented)
+    totalPrice?: number;
+}
+
+// @public
+export interface ReceiptItemArrayField {
+    // (undocumented)
+    type: "array";
+    // (undocumented)
+    valueArray: ReceiptItemField[];
+}
+
+// @public
+export type ReceiptItemField = {
+    type: "object";
+    valueObject: {
+        Name: StringFieldValue;
+        Quantity: NumberFieldValue;
+        TotalPrice: NumberFieldValue;
+    };
+} & CommonFieldValue;
+
+// @public
+export class ReceiptRecognizerClient {
+    constructor(endpointUrl: string, credential: TokenCredential | CognitiveKeyCredential, options?: FormRecognizerClientOptions);
+    readonly endpointUrl: string;
+    // (undocumented)
+    extractReceipt(body: HttpRequestBody, contentType: SupportedContentType, options: StartAnalyzePollerOptions<AnalyzeReceiptResultResponse>): Promise<PollerLike<PollOperationState<AnalyzeReceiptResultResponse>, AnalyzeReceiptResultResponse>>;
+    // (undocumented)
+    extractReceiptFromUrl(imageSourceUrl: string, options: StartAnalyzePollerOptions<AnalyzeReceiptResultResponse>): Promise<PollerLike<PollOperationState<AnalyzeReceiptResultResponse>, AnalyzeReceiptResultResponse>>;
+    }
+
+// @public (undocumented)
+export type ReceiptResult = RawReceiptResult & Receipt;
+
+// @public (undocumented)
+export type StartAnalyzeOptions = ExtractReceiptOptions & {
+    intervalInMs?: number;
+    onProgress?: (state: StartAnalyzePollState<AnalyzeReceiptResultResponse>) => void;
+    resumeFrom?: string;
+};
+
+// @public
+export class StartAnalyzePoller<T extends ICanHazStatus> extends Poller<StartAnalyzePollState<T>, T> {
+    constructor(options: StartAnalyzePollerOptions<T>);
+    // (undocumented)
+    delay(): Promise<void>;
+    // (undocumented)
+    intervalInMs: number;
+}
+
+// @public (undocumented)
+export type StartAnalyzePollerOptions<T extends ICanHazStatus> = {
```

Comment:
yeah I feel like this type shouldn't be public, a dev won't be passing `client` in themselves, or many other of these fields, right? only a subset? I think you need both an internal and an external interface for this.

Reply:
right. consumers don't care about `client`, other may be just needed for the state. I will take another look.

Reply:
Yes, I did have a public-facing option for receipt but didn't use it.  Fixed and updated the other two as well.